subsys/mgmt/Kconfig

```diff
@@ -64,16 +64,77 @@ config MCUMGR_SMP_UART_MTU
 
 source "subsys/mgmt/Kconfig.mcumgr"
 
+config MCUMGR_SMP_UDP
+	bool "UDP mcumgr SMP transport"
+	select MCUMGR
+	select NETWORKING
+	select NET_UDP
+	select NET_SOCKETS
+	select NET_SOCKETS_POSIX_NAMES
+	help
+	  Enables handling of SMP commands received over UDP.
+	  Will start a thread for listening on the configured UDP port.
+
+config MCUMGR_SMP_UDP_IPV4
+	bool "UDP SMP using IPv4"
+	depends on MCUMGR_SMP_UDP
+	depends on NET_IPV4
+	default y
+	help
+	  Enable SMP UDP using IPv4 addressing.
+	  Can be enabled alongside IPv6 addressing.
+
+config MCUMGR_SMP_UDP_IPV6
```

Comment:
Do you need this extra level of configuration? What I mean is that now it is possible to have the device to have IPv6 enabled, but SMP UDP to not support it. Perhaps this is ok, it really depends on the use cases.
For example in networking libs and apps, it is usually enough to just enable IPv6 which then enables IPv6 usage in corresponding lib/app. So usually it is not needed to turn off IPv6 in app if the system globally supports IPv6.

Reply:
It's not necessary, but I figured one might want to disable ipv4/ipv6 for SMP independently of the network stack configuration. I can remove these config items and use the network config values instead if that's preferable.

Reply:
This is more or less up to you and the use cases. If you have a good use case for it, that is fine. It was just not obvious by looking the patch so wanted to clarify it.